### PR TITLE
docs: add doc on state sync configuration

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

This doc is meant to guide a node operator on how to utilize state sync.

Closes: #XXX

### CODE REVIEW

File: docs/tendermint-core/state-sync.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+- `enable`: Enable is to inform the node that you will be using state sync to bootstrap your node.
+- `rpc_servers`: RPC servers are needed because state sync utilizes the light client for verification. 
+    - It is recommended to have more than 2 servers. 
+- `temp_dir`: Temporary directory is store the chunks in the machines local storage, If nothing is set it will create a directory in `/tmp`
+
+The next information you will need to acquire it through publicly exposed RPC's or a block explorer which you trust. 
+
+- `trust_height`: Trust height is needed to inform the light client of your trusted height. 
+- `trust_hash` Trust hash is needed to inform the light client of your trusted hash.
```
Comment (erikgrinaker) [inline docs/tendermint-core/state-sync.md:27]: We should clarify these. `trust_hash` is the block hash of a trusted block (from `block_id.hash` in the header), and `trust_height` is the height of the trusted block.

File: docs/tendermint-core/state-sync.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+```json
+{
+  "height": "273",
+  "hash": "188F4F36CBCD2C91B57509BBF231C777E79B52EE3E0D90D06B1A25EB16E6E23D"
+}
+```
+
+
+When your node is synced and participating in consensus the state sync reactor will still run in the back ground. This is needed to help nodes joining the network get caught up quickly.
```
Comment (erikgrinaker) [inline docs/tendermint-core/state-sync.md:49]: This happens regardless of `enable`, the state sync reactor always runs on all nodes. However, the application has to be configured to take state snapshots - in the Cosmos SDK, this is done by setting `state-sync.snapshot-interval` to a non-zero value.

File: docs/tendermint-core/state-sync.md (hunk 1 above)
Comment (marbar3778) [reply docs/tendermint-core/state-sync.md:27]: Could you not get the hash from multiple places in the RPC. I added an example below on how to get it. Should there be more info in the line for where to get it?

Comment (erikgrinaker) [reply docs/tendermint-core/state-sync.md:27]: Nah, the example is fine. But the descriptions of these options are a little too vague: they're meant to explain what a trusted height and hash are, these just say that they should be the trusted height and hash, so they don't explain anything.

**Review by melekes — APPROVED** [review]
💙

**Pull request merged**